src/test/util/MockMeta.ts

```diff
@@ -0,0 +1,165 @@
+import { Constructor, MetaBase, WidgetMetaConstructor } from '@dojo/framework/widget-core/interfaces';
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import Map from '@dojo/framework/shim/Map';
+
+type MetaFunction = <M extends MetaBase>(MetaType: WidgetMetaConstructor<M>) => any;
+
+export class MockMetaMixin<T extends Constructor<WidgetBase<any>>> {
+	private mockMetaMixin = (Base: T, mockStub: MetaFunction): T => {
+		return class extends Base {
+			private handle: any;
+
+			protected meta<T extends MetaBase>(MetaType: WidgetMetaConstructor<T>): T {
+				return mockStub(MetaType);
+			}
+
+			/**
+			 * Overriding render helps with testing metas that invalidate their host when their values change.
+			 */
+			protected render() {
```

Comment:
I'm slightly confused by this, every time the widget renders it invalidates? This is not representative of a widgets behaviour, effectively forcing/ensuring a widget rerender...

Reply:
Yeah, I am going to update this to allow you to pass an option `shouldInvalidate` when registering meta calls with the `once` function. When it renders, if the next value for that meta has the `shouldInvalidate` it will go ahead and invalidate during the render step. Instead of always invalidating. This will help simulate during tests metas that self invalidate the parent widget.

Reply:
@agubler The PR has been updated to not only invalidate the parent when configured to:

```	it('starts playing on intersection', () => {
		jest.useFakeTimers();
		add('build-time-render', false, true);

		const mockMetaMixin = new MockMetaMixin(GetGoing);
		mockMetaMixin.registerMetaCallOnce(
			Intersection,
			'get',
			['cli'],
			{
				isIntersecting: false
			},
			{
				value: { isIntersecting: true },
				shouldInvalidate: true
			}
		);

		const MockGetGoing = mockMetaMixin.getClass();

		const h = harness(() => <MockGetGoing />);
		h.expect(notPlayingAssertion);

		jest.runAllTimers();

		h.expect(playingAssertion);
	});```